notebooks/train_notebook.ipynb

```diff
@@ -0,0 +1,1144 @@
+            "\tValidation elapsed: 0:1, loss: 0.4187, acc: 90.93%\n",
+            "Epoch 198\n",
+            "\tTrain elapsed: 0:34, loss: 0.0494, acc: 98.41%\n",
+            "\tValidation elapsed: 0:1, loss: 0.4615, acc: 90.53%\n",
+            "Epoch 199\n",
+            "\tTrain elapsed: 0:34, loss: 0.0455, acc: 98.44%\n",
+            "\tValidation elapsed: 0:1, loss: 0.4605, acc: 90.16%\n",
+            "Epoch 200\n",
+            "\tTrain elapsed: 0:34, loss: 0.0513, acc: 98.21%\n",
+            "\tValidation elapsed: 0:1, loss: 0.4459, acc: 91.07%\n"
+          ]
+        }
+      ],
+      "source": [
+        "for epoch in range(1, EPOCHS+1):\n",
+        "    start = time.time()\n",
+        "\n",
+        "    print(f\"Epoch {epoch}\")\n",
+        "    train_loss, train_acc = train_one_epoch(model, train_iterator, criterion, optimizer, device)\n",
+        "    train_mins, train_secs = epoch_time(start, time.time())\n",
+        "\n",
+        "    print(f\"\\tTrain elapsed: {train_mins}:{train_secs}, loss: {train_loss:.4f}, acc: {train_acc * 100:.2f}%\")\n",
+        "\n",
+        "    start = time.time()\n",
+        "    val_loss, val_acc = evaluate(model, valid_iterator, criterion, device)\n",
+        "    val_mins, val_secs = epoch_time(start, time.time())\n",
+        "    scheduler.step()\n",
+        "\n",
+        "    wandb.log({\n",
+        "        \"train_loss\": train_loss,\n",
+        "        \"train_acc\": train_acc,\n",
+        "        \"val_loss\": val_loss,\n",
+        "        \"val_acc\": val_acc,\n",
+        "        \"epoch\": epoch,\n",
+        "        \"current_lr\": scheduler.get_last_lr()[0],\n",
+        "    })\n",
+        "\n",
+        "    print(f\"\\tValidation elapsed: {val_mins}:{val_secs}, loss: {val_loss:.4f}, acc: {val_acc * 100:.2f}%\")\n",
+        "\n",
+        "    if val_loss < best_loss:\n",
+        "        best_loss = val_loss\n",
+        "        torch.save(model.state_dict(), file_path)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "txl2I40ZGwrX"
+      },
+      "source": [
+        "**Evaluate the model**"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 14,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "io_QMFwgGwrX",
+        "outputId": "1c3d6a85-24cb-46a9-8cf3-eb839ae6fdba"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "Test Loss: 0.3867\n",
+            "Test Accuracy: 91.46%\n"
+          ]
+        }
+      ],
+      "source": [
+        "from src.engine import evaluate\n",
+        "test = \"../model/nish_49m_deep6_cyclic_lr_bottleneck.pt\"\n",
+        "model.load_state_dict(torch.load(test))\n",
+        "test_loss, test_acc = evaluate(model.to(device), test_iterator, criterion, device)\n",
```

Comment:
should those two lines just be `model.load_state_dict(torch.load(file_path))`?

Reply:
oops, fixed this